src/modules/fundingManager/oracle/FM_PC_ExternalPrice_Redeeming_v1.sol

```diff
@@ -0,0 +1,698 @@
+    }
+
+    // @inheritdoc IFM_PC_ExternalPrice_Redeeming_v1
+    /// @notice Executes the redemption queue
+    function executeRedemptionQueue()
+        external
+        override(IFM_PC_ExternalPrice_Redeeming_v1)
+        onlyModuleRole(QUEUE_MANAGER_ROLE)
+    {
+        // NOTE: This function expects a Queue-based Payment Processor to be connected.
+        // The call will intentionally revert if a non-Queue PP is used, as this FM
+        // is designed to work only with Payment Processors that support queue-based redemptions.
+        __Module_orchestrator.paymentProcessor().executeRedemptionQueue();
+    }
+
+    //--------------------------------------------------------------------------
+    // Public Functions
+
+    /// @inheritdoc BondingCurveBase_v1
+    /// @dev Only whitelisted addresses can buy
+    /// @param collateralAmount_ The amount of collateral to spend
+    function buy(uint collateralAmount_, uint minAmountOut_)
+        public
+        override(BondingCurveBase_v1)
+        onlyModuleRole(WHITELIST_ROLE)
+    {
+        super.buyFor(_msgSender(), collateralAmount_, minAmountOut_);
+    }
+
+    /// @inheritdoc BondingCurveBase_v1
+    /// @dev Only whitelisted addresses can buy
+    /// @param receiver_ Address to receive tokens
+    /// @param depositAmount_ The amount of collateral to spend
+    /// @param minAmountOut_ The minimum amount of tokens to receive
+    function buyFor(address receiver_, uint depositAmount_, uint minAmountOut_)
+        public
+        override(BondingCurveBase_v1)
+        onlyModuleRole(WHITELIST_ROLE)
+        onlyDirectOperations
+    {
+        super.buyFor(receiver_, depositAmount_, minAmountOut_);
+    }
+
+    /// @inheritdoc RedeemingBondingCurveBase_v1
+    /// @dev Only whitelisted addresses can sell
+    /// @param depositAmount_ Amount of tokens to sell
+    /// @param minAmountOut_ Minimum collateral to receive
+    function sell(uint depositAmount_, uint minAmountOut_)
+        public
+        override(RedeemingBondingCurveBase_v1, IRedeemingBondingCurveBase_v1)
+        onlyModuleRole(WHITELIST_ROLE)
+    {
+        _sellOrder(_msgSender(), depositAmount_, minAmountOut_);
+    }
+
+    /// @inheritdoc RedeemingBondingCurveBase_v1
+    /// @dev Only whitelisted addresses can sell
+    /// @param receiver_ Address to receive tokens
+    /// @param depositAmount_ Amount of tokens to sell
+    /// @param minAmountOut_ Minimum collateral to receive
+    function sellTo(address receiver_, uint depositAmount_, uint minAmountOut_)
+        public
+        override(RedeemingBondingCurveBase_v1, IRedeemingBondingCurveBase_v1)
+        onlyModuleRole(WHITELIST_ROLE)
+        onlyDirectOperations
+    {
+        _sellOrder(receiver_, depositAmount_, minAmountOut_);
+    }
+
+    /// @inheritdoc IFundingManager_v1
+    /// @dev Only payment clients can transfer orchestrator tokens
+    /// @param to_ The recipient address
+    /// @param amount_ The amount to transfer
```

Comment:
The inherited NatSpec already explains this, so this is just double and can be removed.

Reply:
removed